src/patches/key.cc

```diff
@@ -0,0 +1,278 @@
+    {"T", KeyCode::T},
+    {"U", KeyCode::U},
+    {"V", KeyCode::V},
+    {"W", KeyCode::W},
+    {"X", KeyCode::X},
+    {"Y", KeyCode::Y},
+    {"Z", KeyCode::Z},
+    {"NUMLOCK", KeyCode::Numlock},
+    {"KEY0", KeyCode::Keypad0},
+    {"KEY1", KeyCode::Keypad1},
+    {"KEY2", KeyCode::Keypad2},
+    {"KEY3", KeyCode::Keypad3},
+    {"KEY4", KeyCode::Keypad4},
+    {"KEY5", KeyCode::Keypad5},
+    {"KEY6", KeyCode::Keypad6},
+    {"KEY7", KeyCode::Keypad7},
+    {"KEY8", KeyCode::Keypad8},
+    {"KEY9", KeyCode::Keypad9},
+    {"KEYDIVIDE", KeyCode::KeypadDivide},
+    {"KEYENTER", KeyCode::KeypadEnter},
+    {"KEYEQUAL", KeyCode::KeypadEquals},
+    {"KEYMINUS", KeyCode::KeypadMinus},
+    {"KEYMULTI", KeyCode::KeypadMultiply},
+    {"KEYPERIOD", KeyCode::KeypadPeriod},
+    {"KEYPLUS", KeyCode::KeypadPlus},
+};
+
+const std::vector<KeyCode> Key::modifiers = {
+    KeyCode::LeftAlt,
+    KeyCode::LeftApple,
+    KeyCode::LeftCommand,
+    KeyCode::LeftControl,
+    KeyCode::AltGr,
+    KeyCode::LeftShift,
+    KeyCode::LeftWindows,
+    KeyCode::RightAlt,
+    KeyCode::RightApple,
+    KeyCode::RightCommand,
+    KeyCode::RightControl,
+    KeyCode::RightShift,
+    KeyCode::RightWindows,
+};
+
+Key::Key()
+{
+  // Map();
+}
+
+KeyCode Key::Parse(std::string_view key)
+{
+  auto wantedKey = absl::AsciiStrToUpper(key);
+  for (const auto& [value, keycode] : mappedKeys) {
+    if (key == value) {
+      return (KeyCode)keycode;
+    }
+  }
+
+  return KeyCode::None;
+}
+
+bool Key::IsModifier(KeyCode key)
+{
+  return (std::find(Key::modifiers.begin(), Key::modifiers.end(), key) != Key::modifiers.end());
+}
+
+bool Key::IsModified()
+{
+  for (const auto modifier: Key::modifiers) {
+    if (Key::Pressed(modifier)) {
+      return true;
+    }
+  };
+
+  return false;
+}
+
+bool Key::Down(KeyCode key)
+{
+  static auto GetKeyDownInt =
+      il2cpp_resolve_icall<bool(KeyCode)>("UnityEngine.Input::GetKeyDownInt(UnityEngine.KeyCode)");
+  
+  return GetKeyDownInt(key);
+}
+
+bool Key::Pressed(KeyCode key)
+{
+  static auto GetKeyInt = il2cpp_resolve_icall<bool(KeyCode)>("UnityEngine.Input::GetKeyInt(UnityEngine.KeyCode)");
+  
+  return GetKeyInt(key);
+}
+
+bool Key::IsInputFocused()
+{
+  auto eventSystem      = EventSystem::current();
+  if (eventSystem) {
+    auto n = eventSystem->currentSelectedGameObject;
+    if (!n) {
+      return false;
+    }
+    try {
+      if (n) {
+        auto n2 = n->GetComponentFastPath2<TMP_InputField>();
+        if (n2) {
+          return n2->isFocused;
+        }
+      }
+    } catch (...) {
+      return false;
+    }
+  }
+  return false;
+}
+
+bool Key::HasShift()
+{
+  return Key::Pressed(KeyCode::LeftShift) || Key::Pressed(KeyCode::RightShift);
+}
+
+bool Key::HasAlt()
+{
+  return Key::Pressed(KeyCode::LeftAlt) || Key::Pressed(KeyCode::RightShift);
+}
+
+bool Key::HasCtrl()
+{
+  return Key::Pressed(KeyCode::LeftControl) || Key::Pressed(KeyCode::RightControl);
+}
+
+void Key::ClearInput()
```

Comment:
This doesn't really clear, this rather just removes the focus of the input element on the UI tree. This is a workaround for the game not doing that when using keyboard navigation using Escape.
I would suggest renaming this to `ClearInputFocus` or similar to make it clear what the the intent is.

Reply:
Agreed.